metricbeat/beater/reload.go

```diff
@@ -0,0 +1,221 @@
+package beater
+
+import (
+	"expvar"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/pkg/errors"
+
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/logp"
+	"github.com/elastic/beats/libbeat/publisher"
+	"github.com/elastic/beats/metricbeat/mb"
+)
+
+var (
+	debugr        = logp.MakeDebug("reloader")
+	configReloads = expvar.NewInt("metricbeat.config.reloads")
+	moduleStarts  = expvar.NewInt("metricbeat.config.module.starts")
+	moduleStops   = expvar.NewInt("metricbeat.config.module.stops")
+	moduleRunning = expvar.NewInt("metricbeat.config.module.running")
+
+	ErrNoNewFiles = errors.New("no new files found")
+)
+
+type ConfigReloader struct {
+	registry RunningRegistry
+	config   ModulesReloadConfig
+	client   func() publisher.Client
+	done     chan struct{}
+	wg       sync.WaitGroup
+}
+
+type RunningRegistry struct {
```

Comment:
It looks like this is used internally only by the ConfigReloader. Can this be unexported?